### ISSUE TITLE
Feature Request: Suppression of "task created" logs and UI log limitations

### ISSUE THREAD

**Issue by mgmverburg** [PrefectHQ/prefect#5952]
## Description
So, we have been struggling a little with the whole logging situation. We finally had gotten to the point where most things were as we wanted it, until we realized (which was not trivial to realize to begin with) that the UI does not show all the logs. 
At first, we thought, let's see if we can get rid of the default `Created task run 'test_task-c60a5fe2-0' for task 'test_task'` messages by following this: https://discourse.prefect.io/t/how-do-i-suppress-created-task-run-logs/750/6.

However, we actually nicely constructed logger classes in our own codebase that also make use of the Prefect logger, and thus the fix above also affects that, because the feature of the Prefect get_run_logger that it shows which task produced a log seems quite useful, and hence we do not want to lose.

One potential solution could be to allow passing of arguments to get_run_logger such that you can separate the default logs produced by prefect, and logs produced by ourselves, so that more control is given over suppressing the default logs.

Since that is currently not possible, I went back to look another time to see if I could not increase the limit for logs shown (which by default is apparently 200). I had to a deep dive also into the Prefect UI repository to finally figure out that `PREFECT_ORION_API_DEFAULT_LIMIT` was responsible for these limits. This is a quite generically named constant with generic description. 
Perhaps more granularity should somehow be made with regards to the limits, because at the time of writing, I am probably going to increase this limit to 2000, having no idea yet how that will affect other things, since it is a limit that applies to more than just logs.

Basically, I do not think there is any decent workarounds at this point, and it should just be taken into consideration into the design for future releases. We were already considering multiple times to switch to using a different service for logging altogether, and still might, but it is a huge shame because if logs are disfunctional, then Prefect loses a lot of its value for us since a large selling point was its visual UI for more insight into what happens, what goes wrong etc.

**Comment by ahuang11** [comment]
After digging into logging more, I discovered the logging.yml file is pretty powerful, and I was able to remove "Created task run", _without_ dropping other logs!

Here's my `logging.yml` file that I placed in `~/.prefect/logging.yml`
```
# Prefect logging config file.
#
# Any item in this file can be overridden with an environment variable:
#    `PREFECT_LOGGING_[PATH]_[TO]_[KEY]=VALUE`
#
# Templated values can be used to insert values from the Prefect settings at runtime.

version: 1
disable_existing_loggers: False

formatters:
    simple:
        format: "%(asctime)s.%(msecs)03d | %(message)s"
        datefmt: "%H:%M:%S"

    standard:
        format: "%(asctime)s.%(msecs)03d | %(levelname)-7s | %(name)s - %(message)s"
        datefmt: "%H:%M:%S"

    flow_runs:
        format: "%(asctime)s.%(msecs)03d | %(levelname)-7s | Flow run %(flow_run_name)r - %(message)s"
        datefmt: "%H:%M:%S"

    task_runs:
        format: "%(asctime)s.%(msecs)03d | %(levelname)-7s | Task run %(task_run_name)r - %(message)s"
        datefmt: "%H:%M:%S"

    json:
        class: prefect.logging.formatters.JsonFormatter
        format: "default"

filters:
    remove_created_task_run:
        (): __main__.RemoveCreatedTaskRun

handlers:

    # The handlers we define here will output all logs they receieve by default
    # but we include the `level` so it can be overridden by environment

    console:
        level: 0
        class: logging.StreamHandler
        formatter: standard
        filters: [remove_created_task_run]

    console_flow_runs:
        level: 0
        class: logging.StreamHandler
        formatter: flow_runs
        filters: [remove_created_task_run]

    console_task_runs:
        level: 0
        class: logging.StreamHandler
        formatter: task_runs
        filters: [remove_created_task_run]

    orion:
        level: 0
        class: prefect.logging.handlers.OrionHandler

loggers:
    prefect:
        level: "${PREFECT_LOGGING_LEVEL}"
        handlers: [console]
        propagate: no

    prefect.extra:
        level: "${PREFECT_LOGGING_LEVEL}"
        handlers: [orion, console]
        propagate: no

    prefect.flow_runs:
        level: NOTSET
        handlers: [orion, console_flow_runs]
        propagate: no

    prefect.task_runs:
        level: NOTSET
        handlers: [orion, console_task_runs]
        propagate: no

    prefect.orion:
        level: "${PREFECT_LOGGING_SERVER_LEVEL}"

    uvicorn:
        level: "${PREFECT_LOGGING_SERVER_LEVEL}"
        handlers: [console]
        propagate: no

    fastapi:
        level: "${PREFECT_LOGGING_SERVER_LEVEL}"
        handlers: [console]
        propagate: no

# The root logger: any logger without propagation disabled sends to here as well
root:
    # By default, we display warning level logs from any library in the console
    # to match Python's default behavior while formatting logs nicely
    level: WARNING
    handlers: [console]
```

Here's the Python code:
```python
import logging
from prefect import flow, task, get_run_logger


class RemoveCreatedTaskRun(logging.Filter):
    def filter(self, record):
        return 'Created task run' not in record.msg


@flow
def test_flow():
    test_task()


@task
def test_task():
    logger = get_run_logger()
    logger.info("abc")


test_flow()
```

Here's the output!
```
13:00:39.210 | INFO    | prefect.engine - Created flow run 'turquoise-leech' for flow 'test-flow'
13:00:39.211 | INFO    | Flow run 'turquoise-leech' - Using task runner 'ConcurrentTaskRunner'
13:00:39.800 | INFO    | Task run 'test_task-c60a5fe2-0' - abc
13:00:39.931 | INFO    | Task run 'test_task-c60a5fe2-0' - Finished in state Completed()
13:00:40.056 | INFO    | Flow run 'turquoise-leech' - Finished in state Completed('All states completed.')
Completed(message='All states completed.', type=COMPLETED, result=[Completed(message=None, type=COMPLETED, result=None, task_run_id=f448446f-ffd9-48a4-b2bf-b1ab4533af54)], flow_run_id=c189df09-d06d-42d4-90e3-cbe357a7f63d)
```

The possibilities are endless!
```python
class RemoveFinishedInState(logging.Filter):
    def filter(self, record):
        return 'Finished in state' not in record.msg
```
```yaml
    remove_finished_in_state:
        (): __main__.RemoveFinishedInState
     ...
     filters: [remove_created_task_run, remove_finished_in_state]
```

References:
Prefect Logging: https://orion-docs.prefect.io/concepts/logs/#logging-configuration
Logging YAML: https://docs.python.org/3/library/logging.config.html#dictionary-schema-details
Filters: https://stackoverflow.com/questions/61456543/logging-in-python-with-yaml-and-filter

Hope this helps!

**Comment by madkinsz** [comment]
Hi! Thanks for the nice writeup. I'm pretty sure infinite scrolling or retrieval of additional logs in the UI is coming in the future cc @zhen0. We'll also be considering ways to consolidate logs that may be considered "noisy" for easy removal.

**Comment by zhen0** [comment]
My apologies! I missed this one as it didn't have a UI label on it. Infinite scrolling for logs has now been added to the UI flow run page.

**Comment by zhen0** [comment]
Can we consider this one closed or shall we keep open for suppression of minor logs?

**Comment by mgmverburg** [comment]
> My apologies! I missed this one as it didn't have a UI label on it. Infinite scrolling for logs has now been added to the UI flow run page.

Awesome! So there are no more limits to how many logs are displayed? Because indeed we were wondering that with the new release since we saw some changes in the environment variables as well. If that is the case, this can be closed indeed.
